src/accessories/characteristics/targetPosition.ts

```diff
@@ -48,50 +44,59 @@ export class TargetPositionCharacteristic extends TuyaWebCharacteristic {
   ): void {
     const value = (homekitValue as number) === 0 ? 0 : 1;
 
-    const data = { state: value === 0 ? 3 : 1 };
+    const coverAccessory = <CoverAccessory>this.accessory;
+    const target = value ? 100 : 0;
+
+    this.debug("Setting targetPosition to %d", target);
 
     this.accessory
-      .setDeviceState("turnOnOff", { value }, data)
+      .setDeviceState("turnOnOff", { value }, value)
       .then(async () => {
-        this.debug("[SET] %s", value);
+        this.debug("[SET] turnOnOff command sent with value %s", value);
         callback();
 
-        await delay(1000);
-        this.accessory.setTuyaCharacteristic(
+        this.debug("Setting targetPosition to %d", target);
+        coverAccessory.target = target;
+        this.accessory.setCharacteristic(
+          this.accessory.platform.Characteristic.TargetPosition,
+          target,
+          true
+        );
+
+        coverAccessory.motor = value
+          ? this.accessory.platform.Characteristic.PositionState.INCREASING
+          : this.accessory.platform.Characteristic.PositionState.DECREASING;
+        this.accessory.setCharacteristic(
+          this.accessory.platform.Characteristic.PositionState,
+          coverAccessory.motor,
```

Comment:
This will fail when `this.accessory` is an instance of `WindowAccessory` which also uses this characteristic